src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -478,9 +479,11 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             var connectors = obj["Connectors"].ToObject<IEnumerable<ConnectorModel>>(serializer);
 
             var info = new WorkspaceInfo(guid.ToString(), name, description, Dynamo.Models.RunType.Automatic);
-
-            //https://jira.autodesk.com/browse/QNTM-3872 
-            //Category should be set explicitly from custom node workspace.
+            // IsVisibleInDynamoLibrary and Category should be set explicitly for custom node workspace
+            if (obj["View"]["Dynamo"]["IsVisibleInDynamoLibrary"] != null)
+            {
+                info.IsVisibleInDynamoLibrary = obj["View"]["Dynamo"]["IsVisibleInDynamoLibrary"].Value<bool>();
+            }
```

Comment:
is there a test case where "IsVisibleInDynamoLibrary"  is not there in the file? if the other test case handles that ,  I am fine with this change

Reply:
@ramramps The property will be set to true when opening the dyf, or when editing node properties only, it will be saved as false in this case

Reply:
See @mjkkirschner 's comment in the github issue or this line https://github.com/DynamoDS/Dynamo/blob/70bcfb21eb584356c5a4320d060964a6b6a93276/src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs#L206